force-app/main/default/classes/FlowControlLoopsCollections.cls

```diff
@@ -322,17 +428,25 @@ public with sharing class FlowControlLoopsCollections {
      */
     public static Integer getSalary(String employeeName) {
         // Initialize a Map
+        Map<String, Integer> employeesMap = new Map<String, Integer>();
+        List<String> employeesName = new List<String> {'John Doe', 'Jane Smith', 'Sam Brown', 'Alice Johnson'};
+        List<Integer> employeesSalary = new List<Integer> {50000, 60000, 55000, 65000};
+        Integer salary = 0;
 
         // Add key-value pairs to the Map
         // add John Doe => 50000
         // add Jane Smith => 60000
         // add Sam Brown => 55000
         // add Alice Johnson => 65000
+        for (Integer x = 0; x < 4; x++) {
+            employeesMap.put(employeesName[x], employeesSalary[x]);
+        }
 
         // Get the salary of the employee
+        salary = employeesMap.get(employeeName);
 
         // Return the salary of the employee, or null if the employee is not found in the Map
-        return null; // Replace null with the variable you used to store the result
+        return salary; // Replace null with the variable you used to store the result
```

Comment:
See above. You are making your life harder than it has to be. Let me know if you have questions.